extensions/amp-carousel/0.1/scrollable-carousel.js

```diff
@@ -153,30 +149,23 @@ export class AmpScrollableCarousel extends BaseCarousel {
     }
 
     const oldPos = this.pos_;
-
-    const measureSlidePosition = state => {
-      state.newPos = this.getPosForSlideIndex_(index);
-    };
-
-    const mutateSlidePosition = state => {
-      if (state.newPos == oldPos) {
-        return;
-      }
-      /** @const {!TransitionDef<number>} */
-      const interpolate = numeric(oldPos, state.newPos);
-      const duration = 200;
-      const curve = 'ease-in-out';
-      Animation.animate(this.element, pos => {
-        this.container_./*OK*/scrollLeft = interpolate(pos);
-      }, duration, curve).thenAlways(() => {
-        this.commitSwitch_(state.newPos);
-      });
-    };
-
-    this.vsync_.run({
-      measure: measureSlidePosition,
-      mutate: mutateSlidePosition,
-    }, {});
+    this.measureElement(() => this.getPosForSlideIndex_(index))
+        .then(newPos => {
```

Comment:
Note: measure.then.mutate is to take two vsync cycles (~16ms-32ms lag) to complete. You can combine them into a single `#measureMutateElement` call, which will complete in 1 cycle.

Reply:
Will that guarantee that the measure finishes before the mutate? Since I need to use the `index` measured for the mutate.